graphql_compiler/compiler/emit_sql.py

```diff
@@ -1158,14 +1158,23 @@ def unfold(self) -> None:
                 f"Attempted to unfold while the _current_alias was None during fold {self}."
             )
         outer_vertex_primary_key_name = self._get_current_primary_key_name("@fold")
+
+        # Postgres uses a LEFT OUTER JOIN and coalesces nulls to an empty array in the top SELECT.
+        if isinstance(self._sql_schema_info.dialect, PGDialect):
+            is_left_outer_join = True
+        # MSSQL folded subquery returns exactly 1 folded result for each row in the outer table
+        # so should use an INNER JOIN.
+        else:
+            is_left_outer_join = False
```

Comment:
I think most other SQL flavors are likely to be similar to PostgreSQL and not to MSSQL here, so I'd either make the conditional be `if-elif-else raise AssertionError()` or make the `else` case be PostgreSQL + everything else, with MSSQL the "special" if case.

Reply:
Good point! I went with a `NotImplementedError` because that's what we have in other places where we're checking the dialect.